### PR TITLE
Fix code review, add chronos, and shrink images

### PR DESCRIPTION
**Pull request by anhowe**
@ahmetalpbalkan, @rgardler for information

Summary of changes:
- fix all code reviews from @singhkay
- harden shell scripts
- add Chronos support
- update Readme to show how to use Swarm, and how to create Docker applications using Marathon and Chronos

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anhowe__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: mesos-swarm-marathon/configure-mesos-cluster.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@
 ###########################################################
 
 echo "starting mesos cluster configuration"
```
Comment (ahmetalpbalkan) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: Your script does not have set -x or set -e, that's not exactly hardening :-) 

If one of these lines fail, script will keep going. Is that what you intended @anhowe ?

File: mesos-swarm-marathon/configure-mesos-cluster.sh [hunk 2]
```diff
@@ -63,22 +112,63 @@ if isagent ; then
...
+# Start Docker and listen on :2375 (no auth, but in vnet)
+echo 'DOCKER_OPTS="-H unix:///var/run/docker.sock -H 0.0.0.0:2375"' | sudo tee /etc/default/docker
+sudo service docker restart
+
+ensureDocker()
+{
+  # ensure that docker is healthy
+  dockerHealthy=1
+  for i in {1..3}; do
+    sudo docker run hello-world
```
Comment (ahmetalpbalkan) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: You are pulling image down to user's machine and not deleting it. You're not deleting the container earlier. 

I suggest just running "docker info" @anhowe

File: mesos-swarm-marathon/configure-mesos-cluster.sh [hunk 3]
```diff
@@ -109,77 +199,91 @@ if ismaster ; then
...
 ##############################################
 
-echo "restarting mesos processes"
+echo "(re)starting mesos and framework processes"
 if ismaster ; then
   sudo restart zookeeper
   sudo start mesos-master
-  sudo start marathon
+  if [ "$MARATHONENABLED" == "true" ] ; then
+    sudo start marathon
```
Comment (ahmetalpbalkan) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: Not sure what the difference between "service xxx restart" vs "restart xxx" but prefer one for consistency..

I think service command abstracts out init system (upstart vs systemd) might be more portable. @anhowe

File: mesos-swarm-marathon/configure-mesos-cluster.sh (hunk 1 above)
Comment (anhowe) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: There is an explicit abort if the network doesn't come up.  I am holding off on aborting on the remaining errors, as I want to see where and how the failures manifest.

Comment (ahmetalpbalkan) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: Echo statements can fail too. It is a program. I am not saying this script has failure checks. I am saying if something fails in this scrpt, script should exit immediately. Bash default behavior is to continue without exitinf. Please google "bash strict mode" @anhowe

Comment (anhowe) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: As mentioned in previous comment, I want to hold off on fail fast for now, but let's discuss further offline.

File: mesos-swarm-marathon/configure-mesos-cluster.sh (hunk 2 above)
Comment (anhowe) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: Fixed, now using "docker info"

File: mesos-swarm-marathon/configure-mesos-cluster.sh (hunk 3 above)
Comment (anhowe) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: Fixed, now standardized the "service xxx restart" everywhere

File: mesos-swarm-marathon/configure-mesos-cluster.sh (hunk 1 above)
Comment (anhowe) [inline mesos-swarm-marathon/configure-mesos-cluster.sh]: On further discussion, will continue to hold off a bit longer on fast fail (but not ruling out long term) including "set -e" and "set -o pipefail".  Will choose "set -x" as a benefit to see exactly what is being called.

**Pull request merged**